dask/array/overlap.py

```diff
@@ -817,9 +817,7 @@ def coerce_depth(ndim, depth):
     if isinstance(depth, tuple):
         depth = dict(zip(range(ndim), depth))
     if isinstance(depth, dict):
-        for i in range(ndim):
-            if i not in depth:
-                depth[i] = 0
+        depth = {ax: depth.get(ax, default) for ax in range(ndim)}
```

Comment:
The bug is here
https://github.com/dask/dask/blob/1c8b5eafe2be16961472e34fff0c78942f0c6647/dask/array/overlap.py#L527-L531

Iterating over `depth2.values` assumes that `depth2` has keys, values in axes order which it did not.